operations/hpe_pdu/hpe_pdu_admin_procedures.md

```diff
@@ -0,0 +1,108 @@
+#### Update HPE PDU Firmware
+##### Check Firmware Version
+1. Connect to the HPE PDU Web Interface (See [Connect to HPE PDU Web Interface](#connect-to-hpe-pdu-web-interface)) and log in as `admin`
+2. Check which version of firmware is installed by selecting the **"Home"** icon (House in the top right corner) navigate to **"Identification"**
+3. The _"Version"_ will be displayed.  If the version is not _"2.0.0.L"_ update firmware.
+
+##### Update Firmware
+1. Download version **2.0.0.L** firmware from: [https://support.hpe.com/connect/s/search?language=en_US#q=P9S23A&t=All&sort=%40hpescuniversaldate%20descending&numberOfResults=25&f:@contenttype=[Drivers%20and%20Software]](https://support.hpe.com/connect/s/search?language=en_US#q=P9S23A&t=All&sort=%40hpescuniversaldate%20descending&numberOfResults=25&f:@contenttype=[Drivers%20and%20Software])
+This will download an ".exe" file which is a self extracting zip file.
+2. If using a windows system, run the .exe file to extract the files, or use an unzip program on the file.  One of the files extracted will be named **"HPE.FW"**, that is the firmware file you will need for uploading.
+3. Connect to the HPE PDU Web Interface (See [Connect to HPE PDU Web Interface](#connect-to-hpe-pdu-web-interface)) and log in as `admin`.
+4. Using the **"Settings"** Icon (gear in computer monitor in top right corner) navigate to **"System Management"**
+5. Click the **"Update Firmware"** button.
+6. Click **"Choose File"** and select the **"HPE.FW"** file downloaded.
+7. Click **"Upload"** button.
+
+The firmware will be updated and the PDU management processor will restart.
+
+#### Change HPE PDU User Passwords
+1. Connect to the HPE PDU Web Interface (See [Connect to HPE PDU Web Interface](#connect-to-hpe-pdu-web-interface)) and log in as `admin`.
+2. Using the **"admin"** menu (top right corner) and navigate to **"User Accounts"**.
+3. Click on the **"Edit"** icon (pencil) next to the user.
+4. Enter new password and make any other changes for that user account and click the **"Save"** button.
+
+#### Discover HPE PDU after Upgrading CSM
+**This procedure is only needed when upgrading CSM, not performing a fresh install.  This procedure should be run after CSM has been fully upgraded including the discovery job.**
+1.  In CSM 1.0 and before the hms-discovery job and Redfish Translation Service (RTS) treated all PDUs as if were made by ServerTech, and after the upgrade to CSM 1.2 RTS will still think the HPE PDUs in the system are ServerTech PDUs. These erroneous HPE PDU entries for RTS need to get removed from Vault.  
+    ```bash
+    ncn-m001# VAULT_PASSWD=$(kubectl -n vault get secrets cray-vault-unseal-keys -o json | jq -r '.data["vault-root"]' |  base64 -d)
+    ncn-m001# alias vault='kubectl -n vault exec -i cray-vault-0 -c vault -- env VAULT_TOKEN=$VAULT_PASSWD VAULT_ADDR=http://127.0.0.1:8200 VAULT_FORMAT=json vault'
+    ```
+    1.  For each HPE PDU Identified in the command above remove it from Vault:

```

Comment:
I forgot to give you a command. The following should be added before the 2 current sub steps
```
    1.  Identify HPE PDUs that RTS is aware of:
        ```bash
        ncn-m001# vault kv list secret/pdu-creds
        [
          "global/",
          "x3000m0",
          "x3000m1"
        ]
        ```
```

Reply:
Added